third_party/move/move-vm/runtime/src/loader/mod.rs

```diff
@@ -1993,68 +2007,62 @@ impl Loader {
         Ok(match ty {
             Type::Bool => {
                 *count += 1;
-                MoveTypeLayout::Bool
+                (MoveTypeLayout::Bool, false)
             },
             Type::U8 => {
                 *count += 1;
-                MoveTypeLayout::U8
+                (MoveTypeLayout::U8, false)
             },
             Type::U16 => {
                 *count += 1;
-                MoveTypeLayout::U16
+                (MoveTypeLayout::U16, false)
             },
             Type::U32 => {
                 *count += 1;
-                MoveTypeLayout::U32
+                (MoveTypeLayout::U32, false)
             },
             Type::U64 => {
                 *count += 1;
-                MoveTypeLayout::U64
+                (MoveTypeLayout::U64, false)
             },
             Type::U128 => {
                 *count += 1;
-                MoveTypeLayout::U128
+                (MoveTypeLayout::U128, false)
             },
             Type::U256 => {
                 *count += 1;
-                MoveTypeLayout::U256
+                (MoveTypeLayout::U256, false)
             },
             Type::Address => {
                 *count += 1;
-                MoveTypeLayout::Address
+                (MoveTypeLayout::Address, false)
             },
             Type::Signer => {
                 *count += 1;
-                MoveTypeLayout::Signer
+                (MoveTypeLayout::Signer, false)
             },
             Type::Vector(ty) => {
                 *count += 1;
-                MoveTypeLayout::Vector(Box::new(self.type_to_type_layout_impl(
-                    ty,
-                    count,
+                let (layout, has_identifier_mappings) =
+                    self.type_to_type_layout_impl(ty, count, depth + 1)?;
+                (
+                    MoveTypeLayout::Vector(Box::new(layout)),
                     has_identifier_mappings,
-                    depth + 1,
-                )?))
+                )
             },
             Type::Struct { name, .. } => {
                 *count += 1;
-                MoveTypeLayout::Struct(self.struct_name_to_type_layout(
-                    name,
-                    &[],
-                    count,
-                    has_identifier_mappings,
-                    depth,
```

Comment:
is it correct to not use depth+1 here? is this only vectorr depth, but not object depth? should it be renamed?

Reply:
@runtian-zhou

Reply:
But the depth is incremented when processing each field in a struct?